src/adc.rs

```diff
@@ -482,7 +482,7 @@ impl Buffer {
 
         // Let's translate what we got from the DMA peripheral into a write
         // position that we can compare with our read position.
-        let pos = self.len - remaining;
+        let pos = self.len.saturating_sub(remaining) % self.len;
```

Comment:
I don't understand what's happening here. First off, it's very weird that the hardware reports values for `remaining` that are too large, but okay, hardware is often quite weird :-)

Still, it's not clear to me what should happen in that case. The saturating sub will prevent an underflow, which is good, but are we certain that it will lead to correct behavior? Maybe there's an errata note somewhere that can shed some light on the hardware behavior.

I also don't understand what the modulo is doing. Will `self.len.saturating_sub(remaining)` not always result in a value that's smaller or equal to `self.len`? In that case the `% self.len` won't do anything, or am I missing something obvious?